src/collections/vector.js

```diff
@@ -0,0 +1,128 @@
+import * as near from '../api'
+import {u8ArrayToString} from '../utils'
+
+const ERR_INDEX_OUT_OF_BOUNDS = "Index out of bounds"
+const ERR_INCONSISTENT_STATE = "The collection is an inconsistent state. Did previous smart contract execution terminate unexpectedly?"
+
+/// An iterable implementation of vector that stores its content on the trie.
+/// Uses the following map: index -> element
+export class Vector {
+    constructor(prefix) {
+        this.length = 0
+        this.prefix = prefix
+    }
+
+    len() {
+        return this.length
+    }
+
+    isEmpty() {
+        return this.length == 0
+    }
+
+    indexToKey(index) {
+        // TODO: there's no efficient way to convert a Uint64 (BigInt) to Uint8Array
+        // we limit index to be Uint32 for now
+        let data = new Uint32Array([index])
+        let array = new Uint8Array(data.buffer)
+        let key = u8ArrayToString(array)
+        return prefix + key
```

Comment:
shouldn't this be 
```suggestion
        return this.prefix + key
```
or am I misunderstanding this?

Reply:
Good catch! This is how repl manual test can get wrong 😂